### PR TITLE
Added self signup rest api + test case, Added configurations. 

### PR DESCRIPTION
**Pull request by bhathiya**
### Proposed changes in this pull request

Added self signup rest api + test case, Added configurations.

-

### When should this PR be merged

ASAP

### Follow up actions

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java [hunk 1]
```diff
@@ -1301,6 +1293,11 @@ public void removeObserver(EventObserver observer) {
         }
     }
 
+    @Override
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: Add method comments to all public methods except getters and setters.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.wso2.carbon.apimgt.rest.api.store.impl;
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: License header is missing.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    private static final Logger log = LoggerFactory.getLogger(SelfSignupApiServiceImpl.class);
+
+    @Override
+    public Response selfSignupPost(UserDTO body, String contentType, Request request) throws NotFoundException {
+        try {
+            APIStore apiStore = RestApiUtil.getConsumer();
+            apiStore.selfSignUp(MiscMappingUtil.fromUserDTOToUser(body));
+        } catch (APIManagementException e) {
+            String errorMessage = "Error occurred while user signup: " + body.getUsername();
+            HashMap<String, String> paramList = new HashMap<>();
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: use as 

Map<String, String> paramList = new HashMap<>();

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.wso2.msf4j.Request;
+
+import java.util.HashMap;
+import javax.ws.rs.core.Response;
+
+public class SelfSignupApiServiceImpl extends SelfSignupApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(SelfSignupApiServiceImpl.class);
+
+    @Override
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: add method comments

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java (hunk 1 above)
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java]: We agreed to not to put java docs for implemented methods, as they are inherited automatically.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java (hunk 2 above)
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: Fixed.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java (hunk 4 above)
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: This is an autogenerated method.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java (hunk 3 above)
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/main/java/org/wso2/carbon/apimgt/rest/api/store/impl/SelfSignupApiServiceImpl.java]: Fixed.

**Comment by teppi** [comment]
Hi all.
Help me
Step1: I change Label view in file Template.jag( carbon-apimgt-release-6.1.60-i840\features\apimgt\org.wso2.carbon.apimgt.publisher.feature\src\main\resources\publisher\site\themes\wso2\templates\item-design\template.jag) 
Step2: I re-build carbon-apimgt and product-apim úing MAVEN( mvn clean install -Dcheckstyle.skip).
Step3: I extract wso2am-2.1.0.zip in .. \.m2\repository\org\wso2\am\wso2am\2.1.0.
Step4: I run wso2server.bat
But" Label view in file Template.jag" not change.

Can you help me.

Thanks so much

**Pull request merged**